src/include/pagestat_inline.h

```diff
@@ -23,3 +23,61 @@
 
 /* Copy the values from one time page stat structure to another. */
 #define WT_PAGE_STAT_COPY(dest, source) (*(dest) = *(source))
+
+/* Merge a page stat structure into another - summing the value from each. */
+#define WT_PAGE_STAT_MERGE(ps, merge)                                                \
+    do {                                                                             \
+        /* Merge byte count */                                                       \
+        if ((ps)->byte_count == WT_STAT_NONE || (merge)->byte_count == WT_STAT_NONE) \
+            (ps)->byte_count = WT_STAT_NONE;                                         \
+        else                                                                         \
+            (ps)->byte_count += (merge)->byte_count;                                 \
+        /* Merge row count */                                                        \
+        if ((ps)->row_count == WT_STAT_NONE || (merge)->row_count == WT_STAT_NONE)   \
+            (ps)->row_count = WT_STAT_NONE;                                          \
+        else                                                                         \
+            (ps)->row_count += (merge)->row_count;                                   \
+    } while (0)
+
+/*
+ * Update the page stat structure. For both the byte and row counts, we want to reset the page stat
+ * value for the page if any of its children that were merged in had invalid stat values.
+ */
+#define WT_PAGE_STAT_UPDATE(ps, update)                       \
```

Comment:
Is there a case we pass update with -1 count?

Reply:
I think it is possible. When we initialise a `ps` structure it will have -1 values, and it may not get updated to a valid count before calling the macro.